eos/form-factors/parametric-bgl1997-impl.hh

```diff
@@ -33,139 +34,165 @@
-                 UsedParameter(p[_par_name("f_2")],  *this),
-                 UsedParameter(p[_par_name("f_3")],  *this) }},
+    template<typename Process_>
+    BGL1997FormFactors<Process_, PToV>::BGL1997FormFactors(const Parameters & p, const Options & o) :
+        _a_g{{ UsedParameter(p[_par_name("g_0")], *this),
+               UsedParameter(p[_par_name("g_1")], *this),
+               UsedParameter(p[_par_name("g_2")], *this),
+               UsedParameter(p[_par_name("g_3")], *this)
+        }},
+        _a_f{{ UsedParameter(p[_par_name("f_0")], *this),
+               UsedParameter(p[_par_name("f_1")], *this),
+               UsedParameter(p[_par_name("f_2")], *this),
+               UsedParameter(p[_par_name("f_3")], *this)
+        }},
         _a_F1{{  /* F1_0 parameter determined by identity F1(t_-) = (mB - mV) * f(t_-) */
                  UsedParameter(p[_par_name("F1_1")], *this),
                  UsedParameter(p[_par_name("F1_2")], *this),
-                 UsedParameter(p[_par_name("F1_3")], *this) }},
+                 UsedParameter(p[_par_name("F1_3")], *this)
+        }},
         _a_F2{{  /* F2_0 parameter determined by identity between F2 and F1 at q2 = 0 */
                  UsedParameter(p[_par_name("F2_1")], *this),
                  UsedParameter(p[_par_name("F2_2")], *this),
-                 UsedParameter(p[_par_name("F2_3")], *this) }},
+                 UsedParameter(p[_par_name("F2_3")], *this)
+        }},
         _a_T1{{  UsedParameter(p[_par_name("T1_0")], *this),
                  UsedParameter(p[_par_name("T1_1")], *this),
                  UsedParameter(p[_par_name("T1_2")], *this),
-                 UsedParameter(p[_par_name("T1_3")], *this) }},
+                 UsedParameter(p[_par_name("T1_3")], *this)
+        }},
         _a_T2{{ /* T2_0 parameter determined by identity T1(0) = T2(0) */
                  UsedParameter(p[_par_name("T2_1")], *this),
                  UsedParameter(p[_par_name("T2_2")], *this),
-                 UsedParameter(p[_par_name("T2_3")], *this) }},
+                 UsedParameter(p[_par_name("T2_3")], *this)
+        }},
         _a_T23{{ /* T23_0 parameter determined by identity between T2 and T23 at q2 = t_- */
                  UsedParameter(p[_par_name("T23_1")], *this),
                  UsedParameter(p[_par_name("T23_2")], *this),
-                 UsedParameter(p[_par_name("T23_3")], *this) }},
-        _mB(BToDstar::m_B),
-        _mB2(power_of<2>(_mB)),
-        _mV(BToDstar::m_V),
-        _mV2(power_of<2>(_mV)),
-        // default t_0 = sqrt(t_p) (sqrt(m_B) - sqrt(m_M))^2 (optimal value)
-        _t_0(p["B->D^*::t_0@BGL1997"], *this)
+                 UsedParameter(p[_par_name("T23_3")], *this)
+        }},
+        _traits(BGL1997FormFactorTraits<Process_, PToV>(p)),
+        _mB(_traits.m_B),
+        _mV(_traits.m_V),
+        t_0(_traits.t_0),
+        n_bound_states_1m(o, _options, "n-bound-states-1m"),
```

Comment:
As discussed in the meeting this morning: it is preferable to move this logic into the FormFactorTraits class and let it provide four Blaschke factor functions, one for each possible number of resonances. Then one ``std::function`` object per channel can be constructed here to take of each of the four channels' Blaschke factors.